src/Storages/Kafka/KafkaConfigLoader.cpp

```diff
@@ -160,6 +170,15 @@ namespace
 
 void setKafkaConfigValue(cppkafka::Configuration & kafka_config, const String & key, const String & value)
 {
+    /// Skip ClickHouse internal settings that are read directly in code and should not be passed to librdkafka
+    static const std::unordered_set<String> clickhouse_internal_settings = {
+        "use_environment_credentials",  // AWS MSK IAM: controls credential source (env vs instance profile)
+        // Add future internal settings here
+    };
```

Comment:
I like the general approach (have a list of internal settings that we ignore), but let's ignore them in `loadFromConfig`, similarly how `CONFIG_KAFKA_PRODUCER_TAG` and `CONFIG_KAFKA_CONSUMER_TAG` are ignored.